### PR TITLE
Added markup chapter

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Fixed tickets | none
| License | MIT

#### What's in this PR?

This PR adds the markup chapter

#### Why?

Because of the feature which will be introduced by https://github.com/sulu/sulu/pull/2338

### CODE REVIEW

**Comment by danrot** [comment]
This PR contains too much, the plugin section for the ckeditor should be a separate PR.

**Comment by wachterjohannes** [comment]
@danrot i have extracted the commit for ckeditor plugins https://github.com/sulu/sulu-docs/pull/224

File: bundles/markup/index.rst [hunk 1]
```diff
@@ -0,0 +1,72 @@
+MarkupBundle
+============
+
+The MarkupBundle provides the feature of declaring different "relations" as
```
Comment (danrot) [inline bundles/markup/index.rst]: Relations doesn't make sense here IMO.

File: bundles/markup/index.rst [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+html-tag. This tags will be automatically parsed before the response will be
```
Comment (danrot) [inline bundles/markup/index.rst]: `These tags`

File: bundles/markup/index.rst [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+send.
```
Comment (danrot) [inline bundles/markup/index.rst]: sent

File: bundles/markup/index.rst [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+Example
+-------
+
+This example contains a sulu-related example. The tag ``sulu:link`` represents
+a link to another page. This tag will be replaces via a valid anchor where the
```
Comment (danrot) [inline bundles/markup/index.rst]: `will be replaced`

File: bundles/markup/index.rst [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+href links to the page with given UUID.
```
Comment (danrot) [inline bundles/markup/index.rst]: `href attribute contains the UUID of the page`, and put `href` in double backticks.

File: bundles/markup/index.rst [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+.. toctree::
+    :maxdepth: 1
+
+    link
+
+Extending
+---------
+
+To enable replacement of your "custom" tags you can define a service which
```
Comment (danrot) [inline bundles/markup/index.rst]: Do not use quotes here.

File: bundles/markup/index.rst [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+         *
+         * @return array Tag array to replace all occurrences.
+         */
+        public function parseAll($attributesByTag) {
+            $result = [];
+            foreach($attributesByTag as $tag => $attributes) {
+                $url = ; // load url via uuid from document-manager
+                $pageTitle = ...; // load page-title via uuid from document-manager
+
+                $result[$tag] = sprintf('<a href="%s" title="%s">%s</a>', $url, $$pageTitle, $attributes['content']);
```
Comment (danrot) [inline bundles/markup/index.rst]: `$pageTitle`, not `$$pageTitle`

**Comment by wachterjohannes** [comment]
@danrot fixed

**Pull request merged**